lib/stat_tracker.rb

```diff
@@ -315,9 +315,57 @@ def worst_coach(season)
     coach_losses.max_by { |coach| losses_by_coach[coach] }
   end
 
-  # def most_accurate_team(season)
-  #
-  # end
+  def most_accurate_team(season)
+    season_game_ids = season_games(season).map do |game|
+      game.game_id
+    end
+
+    season_games = game_teams.find_all do |game|
+    season_game_ids.include?(game.game_id)
+    end
+
+    season_shots_by_team = season_games.inject(Hash.new(0)) do |season_team_shots, game|
+      season_team_shots[game.team_id] += game.shots.to_i
+      season_team_shots
+    end
+
+    season_goals_by_team = season_games.inject(Hash.new(0)) do |season_team_goals, game|
+      season_team_goals[game.team_id] += game.goals.to_i
+      season_team_goals
+    end
+
+    season_team_accuracy = {}
+    season_goals_by_team.each do |team_goals_id, goals|
+      season_shots_by_team.each do |team_shots_id, shots|
+        if team_goals_id == team_shots_id
```

Comment:
@sciencefixion i find this conditional statement problematic because your code below only runs if the # of goals == # of shots, but that wouldn't be the case, right? We'd have more shots than goals? So maybe you don't need a conditional at all, you just want to do the math that you wrote below, I think.

Reply:
My variables could use better naming I think. I am trying take my hashes of season_team_goals and season_team_shots and produce a new hash that has the accuracy percentage. I obviously struggled here because I have nested iteration. But I'm trying to match the keys so my math is working on the right numbers and my result is still connected to the team_id.

TL;DR The conditional is there to match the two keys in separate hashes. The result should be a new has with team_id => accuracy percentage.